### ISSUE TITLE
Improve `deploy:activate`

### ISSUE THREAD

**Issue by quiddle** [Kerry350/ember-deploy-s3-index#5]
I've been considering putting together a PR that improves the behavior of deploy:activate.

My thoughts:
1. `ember deploy:activate` with no arguments should activate the most recent version, simplifying the process.
2. Allow passing the numbers displayed in `deploy:list` as arguments to `deploy:activate` 

``` sh
$ ember deploy:list 
version: 0.1.15
Found the following revisions: 

1) my-app:5a77e50 
2) my-app:b8256de 
3) my-app:c239613  

Use activate() to activate one of these revisions

$ ember deploy:activate 
# would activate my-app:5a77e50

$ ember deploy: activate 2
# would activate  my-app:b8256de
```

Do you think these would be useful features?

**Comment by LevelbossMike** [comment]
I think this is something that should to be changed in ember-cli-deploy and not in the adapters directly if we want to do this.

**Comment by Kerry350** [comment]
@quiddle I like the ideas :+1:  I agree with @LevelbossMike though that this should move across `ember-cli-deploy` itself if there's interest for it, or else the adapters will diverge with their behaviour.

**Comment by quiddle** [comment]
Hmm…. Let's implement it wherever it works best!

@LevelbossMike Unless I'm mistaken, ember-cli-deploy doesn't know anything about the revisions themselves  (or whether they are labeled with numbers or letters or roman numerals) and leaves displaying the revisions up to the index adapter. In order to standardize on this type of behavior, I'd think that ember-cli-deploy would need to receive an array of revisions and then handle displaying them. PR #4 in this repo has some improvements I found to be useful when displaying the list of revisions, that I'd be happy to bring over to ember-cli-deploy if we want to make those kinds of changes.

**Comment by andremalan** [comment]
One thought on "deploy:activate" is right now it copies the revision file over to index.html, making it hard to know what the current active revision is. 

Is there a reason it does that instead of updating the "website index document" setting of the bucket to be the active revision file?

**Comment by quiddle** [comment]
@andremalan I think that's definitely something that needs handled. I put together a PR (#4) that adds an extra file for keeping track of the current revision.

Being able to change the 'website index document' would work if you were hosting your index file directly from your S3 bucket. Take a look at what's going on over at ember-cli/ember-cli-deploy to see how the adapter/plugin story is changing. It looks like you'd be able to create a solution the way you want.

**Comment by andremalan** [comment]
@quiddle, yeah, I see the other PR and the discussions in ember-cli-deploy. I'm more worried about in the context of an S3 only deploy is there ever a case where updating the index in the web hosting settings doesn't make sense.

If we don't think there is such a case, then I can add some thoughts to your PR about using a file vs. the index.

**Comment by quiddle** [comment]
If you're doing direct s3 hosting, I don't see a reason why not, but perhaps someone else can chime in. It might be possible to use a config option on the adapter/plugin to change behavior.
